bundles/org.openhab.binding.vesync/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,343 @@
+
+	<bridge-type id="bridge">
+		<label>Vesync Bridge</label>
+		<description>The Vesync bridge represents the Vesync cloud service.</description>
+
+		<properties>
+			<property name="Registration Time"/>
+			<property name="Country Code"/>
+			<property name="Accept Language"/>
+		</properties>
+
+		<config-description>
+			<parameter name="username" type="text">
+				<context>password</context>
+				<label>Username</label>
+				<description>Name of a registered VeSync user, that allows to access the mobile application.</description>
+			</parameter>
+			<parameter name="password" type="text">
+				<context>password</context>
+				<label>Password</label>
+				<description>Password for the registered VeSync username, that allows to access the mobile application.</description>
+			</parameter>
+			<parameter name="backgroundDeviceDiscovery" type="boolean">
+				<label>Enable Background Device Scans</label>
+				<description>Enable background scanning for new devices.</description>
+				<default>false</default>
+			</parameter>
+			<parameter name="airPurifierPollInterval" type="integer" min="5" step="1" unit="s">
+				<label>Polling Interval For Air Filters / Humidifiers</label>
+				<description>Seconds between fetching background updates about the air purifiers / humidifiers.</description>
+				<default>5</default>
+			</parameter>
+		</config-description>
+	</bridge-type>
+
+	<thing-type id="AirPurifier">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+
+		<label>Air Purifier via VeSync</label>
+		<description>A Air Purifier uplinking to VeSync</description>
+
+		<channels>
+			<channel id="enabled" typeId="deviceEnabledType"/>
+			<channel id="childLock" typeId="deviceChildLockEnabledType"/>
+			<channel id="display" typeId="deviceDisplayEnabledType"/>
+			<channel id="filterLifePercentage" typeId="deviceFilterLifePercentageType"/>
+			<channel id="fanMode" typeId="airPurifierModeType"/>
+			<channel id="manualFanSpeed" typeId="airPurifierFanLevelType"/>
+			<channel id="errorCode" typeId="deviceErrorCodeType"/>
+			<channel id="airQuality" typeId="deviceAirQualityBasicType"/>
+			<channel id="airQualityPPM25" typeId="deviceAirQualityBasicType"/>
+			<channel id="configDisplayForever" typeId="deviceAFConfigDisplayForever"/>
+			<channel id="configAutoMode" typeId="deviceAFConfigAutoPrefType"/>
+			<channel id="timerExpiry" typeId="deviceAFTimerExpiry"/>
+			<channel id="configAutoRoomSize" typeId="deviceAFConfigAutoPrefRoomSizeType"/>
+			<channel id="schedulesCount" typeId="deviceAFConfigAutoScheduleCountType"/>
+			<channel id="nightLightMode" typeId="deviceAFNightLight"/>
+		</channels>
+
+		<properties>
+			<property name="Device Name"/>
+			<property name="Device Type"/>
+			<property name="MAC Id"/>
+		</properties>
+		<representation-property>mac</representation-property>
+
+		<config-description>
+			<parameter name="macId" type="text">
+				<context>text</context>
```

Comment:
Are you sure this conext exists? It isn't documented here https://www.openhab.org/docs/developer/bindings/config-xml.html#xml-structure-for-configuration-descriptions

Please check all.

Reply:
Thanks - sorry that's a good ref to bookmark :) Not sure where I pulled that from :( I've updated the bridge configuration while looking through this context request.